### PR TITLE
Change default value of test property CodeActionValidationMode to SemanticStructure instead of None and update so that tests still pass

### LINKED ISSUE #2785: Avoid using CodeFixValidationMode.None in tests
🔗 Issue filed in response to https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/2751#discussion_r215483408

* Tests should validate the semantic structure by default
* The few tests which fail validation should be updated to pass semantic structure validation or use a localized configuration of `CodeFixValidationMode.None`

### PR DESCRIPTION
**Pull request by bjornhellander**
Fixes #2785

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/StyleCop.Analyzers.CodeFixes.csproj [hunk 1]
```diff
@@ -36,6 +36,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="1.2.1" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="1.2.1" />
+    <PackageReference Include="System.ValueTuple" Version="4.5.0" />
```
Comment (bjornhellander) [inline StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/StyleCop.Analyzers.CodeFixes.csproj:39]: Is this ok?

Comment (sharwell) [reply StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/StyleCop.Analyzers.CodeFixes.csproj:39]: I'd be concerned about the runtime impact of this on VS 2015. It seems like we could use `out` parameters instead pretty easily.

Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/StyleCop.Analyzers.CodeFixes.csproj:39]: Absolutely. I will update.

Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/StyleCop.Analyzers.CodeFixes.csproj]: Updated

**Review by sharwell — APPROVED** [review]

**Pull request merged**